src/components/dialogs/network-modifications/generator/modification/regulating-terminal-modification-dialog.js

```diff
@@ -0,0 +1,174 @@
+import { Grid } from '@mui/material';
+import RegulatingTerminalForm from 'components/dialogs/regulating-terminal/regulating-terminal-form';
+import { getTapChangerEquipmentSectionTypeValue } from 'components/utils/utils';
+import { fetchVoltageLevelsListInfos } from 'services/study/network';
+import { getRegulatingTerminalFormData } from 'components/dialogs/regulating-terminal/regulating-terminal-form-utils';
+
+const emptyFormData = {
+    [VOLTAGE_LEVEL]: null,
+    [EQUIPMENT]: null,
+};
+
+const RegulatingTerminalModificationDialog = ({
+    data,
+    previousData,
+    currentNode,
+    studyUuid,
+    onModifyRegulatingTerminalGenerator,
+    ...dialogProps
+}) => {
+    const formSchema = yup
+        .object()
+        .shape({
+            [VOLTAGE_LEVEL]: yup
+                .object()
+                .nullable()
+                .shape({
+                    [ID]: yup.string(),
+                    [NAME]: yup.string(),
+                    [SUBSTATION_ID]: yup.string(),
+                    [NOMINAL_VOLTAGE]: yup.string(),
+                    [TOPOLOGY_KIND]: yup.string().nullable(),
+                })
+                .required(),
+            [EQUIPMENT]: yup
+                .object()
+                .nullable()
+                .shape({
+                    [ID]: yup.string(),
+                    [NAME]: yup.string().nullable(),
+                    [TYPE]: yup.string(),
+                })
+                .required(),
+        })
+        .required();
+    const formMethods = useForm({
+        defaultValues: emptyFormData,
+        resolver: yupResolver(formSchema),
+    });
+
+    const { reset } = formMethods;
+    const clear = useCallback(() => {
+        reset(emptyFormData);
+    }, [reset]);
+
+    const [voltageLevelOptions, setVoltageLevelOptions] = useState([]);
+
+    useEffect(() => {
+        if (studyUuid && currentNode.id) {
+            fetchVoltageLevelsListInfos(studyUuid, currentNode.id).then(
+                (values) => {
+                    setVoltageLevelOptions(
+                        values.sort((a, b) => a.id.localeCompare(b.id))
+                    );
+                }
+            );
+        }
+    }, [studyUuid, currentNode]);
+
+    const onSubmit = useCallback(
+        (voltageRegulationGenerator) => {
+            onModifyRegulatingTerminalGenerator(voltageRegulationGenerator);
+        },
+        [onModifyRegulatingTerminalGenerator]
+    );
+
+    const fromEditDataToFormValues = useCallback(
+        (data) => {
+            reset({
+                ...getRegulatingTerminalFormData({
+                    equipmentId: data.regulatingTerminalConnectableId,
+                    equipmentType: data.regulatingTerminalConnectableType,
+                    voltageLevelId: data.regulatingTerminalVlId,
+                }),
+            });
+        },
+        [reset]
+    );
+
+    useEffect(() => {
+        if (data) {
+            if (data.regulatingTerminalConnectableId !== ' ') {
```

Comment:
what if `data.regulatingTerminalConnectableId === ''` or 
`data.regulatingTerminalConnectableId === '   '` (n number of white spaces)